### PR TITLE
Task #1459163: Write tests for FileFormatFactory

### PR DESCRIPTION
**Pull request by assapir**

### CODE REVIEW

File: sdk/file_sdk/UnitTests/common_ut/common_ut.cpp [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+  }
+
+  void Create_UknonwExtenstion_ReturnPFileFileFormat();
+};
+
+CommonTests::CommonTests()
+{
+}
+
+void CommonTests::Create_StreamIsNull_ThrowInvlidArgument()
```
Comment (vakuras) [inline sdk/file_sdk/UnitTests/common_ut/common_ut.cpp]: typo #Resolved

File: sdk/file_sdk/UnitTests/common_ut/common_ut.cpp [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+  QVERIFY_EXCEPTION_THROWN(FileFormatFactory::Create(nullptr, ""), std::invalid_argument);
+}
+
+void CommonTests::Create_NoExtension_ThrowInvalidArgument()
+{
+  auto stream = std::make_shared<IStreamMock>();
+  QVERIFY_EXCEPTION_THROWN(FileFormatFactory::Create(stream, ""), std::invalid_argument);
+}
+
+void CommonTests::Create_ExtenstionWithoutDot_AddsDot()
```
Comment (vakuras) [inline sdk/file_sdk/UnitTests/common_ut/common_ut.cpp]: what about the case of passing .xxx.pfile? #Resolved

File: sdk/file_sdk/UnitTests/common_ut/common_ut.cpp [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+{
+  QFETCH(std::string, extenstion);
+  auto stream = std::make_shared<IStreamMock>();
+  auto fileFormat = FileFormatFactory::Create(stream, extenstion);
+  QVERIFY2(dynamic_cast<XMPFileFormat*>(fileFormat.get()) != NULL, "is not XMPFileFormat");
+}
+
+void CommonTests::Create_CompoundFileExtenstion_ReturnCompoundFileFormat()
+{
+  QFETCH(std::string, extenstion);
```
Comment (vakuras) [inline sdk/file_sdk/UnitTests/common_ut/common_ut.cpp]: typo #Resolved

**Review by vakuras — CHANGES_REQUESTED** [review]
:clock1:

File: sdk/file_sdk/Common/file_format_factory.cpp [hunk 4]
```diff
@@ -40,15 +40,26 @@ bool FileFormatFactory::in_array(const std::string &value, const vector<std::str
...
     throw std::invalid_argument("The extention is empty");
   }
 
   std::string finalExtention = extension;
-  if(extension.find_last_of('.') == string::npos) {
+  auto pos = extension.find_last_of('.');
+  if(pos == string::npos) {
     // TODO: add log
     finalExtention.insert(finalExtention.begin(), '.');
+    pos = finalExtention.find_last_of('.');
```
Comment (vakuras) [inline sdk/file_sdk/Common/file_format_factory.cpp]: this line exist only for the point of not getting into the next if #Resolved

**Review by vakuras — APPROVED** [review]
:shipit:

File: sdk/file_sdk/UnitTests/common_ut/istream_mock.cpp [hunk 5]
```diff
@@ -0,0 +1,69 @@
+#include "istream_mock.h"
+
+IStreamMock::IStreamMock()
```
Comment (shenhavba) [inline sdk/file_sdk/UnitTests/common_ut/istream_mock.cpp:3]: the "{" in this file are not in the same line #Resolved

**Review by shenhavba — APPROVED** [review]
:shipit:

**Pull request merged**